core/src/main/java/io/jenkins/pluginhealth/scoring/probes/SCMLinkValidationProbe.java

```diff
@@ -76,25 +87,62 @@ protected boolean requiresRelease() {
         return true;
     }
 
-    private ProbeResult fromSCMLink(ProbeContext context, String scm) {
+    private ProbeResult fromSCMLink(ProbeContext context, String scm, String pluginName) {
         Matcher matcher = GH_PATTERN.matcher(scm);
         if (!matcher.find()) {
             if (LOGGER.isDebugEnabled()) {
-                LOGGER.debug("{} is not respecting the SCM URL Template", scm);
+                LOGGER.debug("{} is not respecting the SCM URL Template.", scm);
             }
-            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories");
+            return ProbeResult.failure(key(), "SCM link doesn't match GitHub plugin repositories.");
         }
-
         try {
             context.getGitHub().getRepository(matcher.group("repo"));
-            return ProbeResult.success(key(), "The plugin SCM link is valid");
+            Optional<Path> file = searchPomFiles(context.getScmRepository(), pluginName);
+            String folderPath = file.isPresent() ? new File(file.toString()).getParentFile().getName() : "";
+            context.setScmFolderPath(folderPath);
+            return ProbeResult.success(key(), "The plugin SCM link is valid.");
         } catch (IOException ex) {
-            return ProbeResult.failure(key(), "The plugin SCM link is invalid");
+            return ProbeResult.failure(key(), "The plugin SCM link is invalid.");
         }
     }
 
     @Override
     public String[] getProbeResultRequirement() {
         return new String[]{UpdateCenterPluginPublicationProbe.KEY};
     }
+
+    /**
+     * Searches for Pom files in every directory available in the repository
+     *
+     * @param directory  in the scm
+     * @param pluginName the name of the plugin
+     * @return folderPath if it valid or return the scm itself
+     */
+    private Optional<Path> searchPomFiles(Path directory, String pluginName) {
+        Optional<Path> fileName = Optional.empty();
+        try (Stream<Path> paths = Files.find(directory, 2, (path, $) ->
+            path.getFileName().toString().equals("pom.xml"))) {
+            fileName = paths
```

Comment:
You can directly 

`return directories.filter(pom -> pomFileMatchesPlugin(pom, pluginName)).findFirst();`

instead of having that assigned to variables. If you read the docs of `findFirst`, it tells you that it'll return an empty `Optional` if nothing is found, there is no need to repeat that yourself.